src/core/SamplePlayHandle.cpp

```diff
@@ -32,25 +32,26 @@
 
 
 
-SamplePlayHandle::SamplePlayHandle( SampleBuffer* sampleBuffer ) :
+SamplePlayHandle::SamplePlayHandle( SampleBuffer* sampleBuffer , bool shouldCreateAudioPort) :
 	PlayHandle( TypeSamplePlayHandle ),
 	m_sampleBuffer( sharedObject::ref( sampleBuffer ) ),
 	m_doneMayReturnTrue( true ),
 	m_frame( 0 ),
-	m_ownAudioPort( true ),
+	m_ownAudioPort( shouldCreateAudioPort ),
 	m_defaultVolumeModel( DefaultVolume, MinVolume, MaxVolume, 1 ),
 	m_volumeModel( &m_defaultVolumeModel ),
 	m_track( NULL ),
 	m_bbTrack( NULL )
 {
-	setAudioPort( new AudioPort( "SamplePlayHandle", false ) );
+	if (shouldCreateAudioPort)
+		setAudioPort( new AudioPort( "SamplePlayHandle", false ) );
 }
 
 
 
 
 SamplePlayHandle::SamplePlayHandle( const QString& sampleFile ) :
-	SamplePlayHandle( new SampleBuffer( sampleFile ) )
+	SamplePlayHandle( new SampleBuffer( sampleFile ) , false)
```

Comment:
According tohttps://github.com/LMMS/lmms/pull/4310/files#diff-154de9db984f4e39bfeb6e62700ecc83L40, `true` should be passed.